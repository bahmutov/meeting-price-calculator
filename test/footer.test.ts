import xs from 'xstream';
import { VNode } from '@cycle/dom';
import { withTime } from 'cyclejs-test-helpers';
const htmlLooksLike = require('html-looks-like');
const toHtml = require('snabbdom-to-html');

import Footer from '../src/components/footer';

describe('Footer Component', () => {
  it('should render correctly', () => {
    const expectedHTML = `
      <div class="Footer">
        <a class="Footer-link"
            href="https://cycle.js.org/"
            target="_blank">
            <span class="Footer-label">Built with CycleJS</span>
            <div class="Footer-cycle-logo"></div>
        </a>
        <a class="Footer-link"
            href="https://github.com/olpeh/meeting-price-calculator"
            target="_blank">
          <span class="Footer-label">By olpeh</span>
          <div class="Footer-github-logo"></div>
        </a>
      </div>
    `;

    const vdom$ = Footer();
    const html$ = vdom$.map(toHtml);
    const expected$ = xs.of(expectedHTML);
    const looksLike = withTime(Time =>
      Time.assertEqual(html$, expected$, htmlLooksLike)
    );
    return looksLike();
  });

  it('should match a snapshot correctly', () => {
    const vdom$: xs<VNode> = Footer();
    const html$: xs<{}> = vdom$.map(toHtml);
    expect(html$).toMatchSnapshot();
  });
});
