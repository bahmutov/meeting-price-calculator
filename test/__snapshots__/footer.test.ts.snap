// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match a snapshot correctly 1`] = `
Stream {
  "_d": false,
  "_dl": Object {},
  "_err": Object {},
  "_ils": Array [],
  "_prod": MapOp {
    "f": [Function],
    "ins": Stream {
      "_d": false,
      "_dl": Object {},
      "_err": Object {},
      "_ils": Array [],
      "_prod": FromArray {
        "a": Array [
          <footer.footer_f19tl4e9>
            <a.link_frxg3q8
              attrs={
                Object {
                  "href": "https://cycle.js.org/",
                  "target": "_blank",
                }
              }
            >
              Built with CycleJS
              <img.logo_f7jvbo9
                attrs={
                  Object {
                    "src": Object {
                      "process": [Function],
                    },
                  }
                }
              />
            </a.link_frxg3q8>
            <a.link_frxg3q8
              attrs={
                Object {
                  "href": "https://github.com/olpeh/meeting-price-calculator",
                  "target": "_blank",
                }
              }
            >
              By olpeh
              <img.logo_f7jvbo9
                attrs={
                  Object {
                    "src": Object {
                      "process": [Function],
                    },
                  }
                }
              />
            </a.link_frxg3q8>
          </footer.footer_f19tl4e9>,
        ],
        "type": "fromArray",
      },
      "_stopID": Object {},
      "_target": Object {},
    },
    "out": Object {},
    "type": "map",
  },
  "_stopID": Object {},
  "_target": Object {},
}
`;
